src/Features/Core/Portable/ExtractClass/AbstractExtractClassRefactoringProvider.cs

```diff
@@ -24,15 +24,7 @@ internal abstract class AbstractExtractClassRefactoringProvider(IExtractClassOpt
 
     public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
     {
-        // For simplicity if we can't add a document the don't supply this refactoring. Not checking this results in known
-        // cases that won't work because the refactoring may try to add a document. There's non-trivial
-        // work to support a user interaction that makes sense for those cases. 
-        // See: https://github.com/dotnet/roslyn/issues/50868
         var solution = context.Document.Project.Solution;
-        if (!solution.CanApplyChange(ApplyChangesKind.AddDocument))
```

Comment:
This will now be handled by disabling the new file option on the extract dialog.